core/services/relay/evm/codec_entry.go

```diff
@@ -42,22 +42,30 @@ func (entry *codecEntry) Init() error {
 		return nil
 	}
 
+	numIndecies := 0
 	seenNames := map[string]bool{}
 	for i, arg := range args {
-		tmp := arg.Type
-		nativeArg, checkedArg, err := getNativeAndCheckedTypes(&tmp)
+		if arg.Indexed {
+			if numIndecies == 3 {
+				return fmt.Errorf("%w: too many indexed arguments", commontypes.ErrInvalidConfig)
+			}
+			numIndecies++
+		}
+
+		tmp := arg
+		nativeArg, checkedArg, err := getNativeAndCheckedTypesForArg(&tmp)
 		if err != nil {
 			return err
 		}
 		if len(arg.Name) == 0 {
```

Comment:
Why this instead of == "" ? Also won't event topics fail here because the first one(signature) doesn't have a name

Reply:
The first (and only arg) not having a name, but having subfields indexed isn't supported.  I can comment that go-ethereum wouldn't really support it correctly.

Reply:
what do you mean by the first part of why not ==?  := can't be replaced with ==.  If you meant why not =, you need temporary variables in a loop or the variable you point to will change on each iteration.

Reply:
I meant this part `if len(arg.Name) == 0 ` can be `if arg.Name == "" `

Reply:
wdym by subfields do you mean this?
```
function foo() returns int256
function foo() returns ( int256, j int256)
```

Doesn't this same function also encode log event topics which I think have the first topic unnamed even when there are multiple

Reply:
> `I meant this part if len(arg.Name) == 0  can be if arg.Name == "" `


Sure, I imagine length checks are cheaper though (the compiler might optimize this case into the len check, not sure)

> `function foo() returns ( int256, j int256)
`
The unnamed argument wouldn't be able to be decoded into a struct (or json or a map), so I think we would just drop it.  If you had two unnamed, it would return an error because they share a name.  It might cause a panic right now because of an unnamed struct field... I can add a test case to make sure it should return an error.